reference/hoon-expressions/rune/bar.md

```diff
@@ -8,12 +8,78 @@ Core expressions produce cores. A core is a cell of `[battery payload]`.
 The `battery` is code, a battery of Nock formulas.  The `payload` is the data
 needed to run those formulas correctly.
 
-Four core runes (`|=`, `|.`, `|-`, and `|*`) produce a
+Five core runes (`|=`, `|.`, `|-`, `|*`, and `|$`) produce a
 core with a single arm, named `$`. We can recompute this arm
-with changes, which useful for recursion among other things:
+with changes, which is useful for recursion among other things.
 
 ## Runes
 
+### |$ "barbuc"
+
+Declares a mold builder wet gate with one or more molds as its
+sample.
+
+##### Syntax
+
+Regular:
+```hoon
+|$  a=(lest term)  b=spec
+```
+
+AST:
+```hoon
+[%brbs sample=(lest term) body=spec]
+```
+
+##### Semantics
+
+`|$` is used to declare a wet gate that is a mold builder. The mold builder
+is polymorphic in its input types.  `a` is a `lest` of `term` used as
+identifiers for the input types. `b` is a structure built from elements of `a`.
+The output of `|$` is a mold builder that produces a `spec` obtained by
+substituting the input types parameterized by `a` into `b`.
```

Comment:
A `$spec` is the usual compile-time representation of a mold. A mold-builder produces a mold, not a spec.

Reply:
Thank you! I have been confused about this since forever. Also rolled this into https://github.com/urbit/docs/pull/877/commits/1a0e1a4616736a323a80eddc823a049b9cfe0cf0 which is the same commit for the above convo